test/e2e/e2e_test.go

```diff
@@ -99,7 +111,7 @@ var _ = Describe("e2e control plane", func() {
 		podList, _ := podClient.List(metav1.ListOptions{})
 		podName := ""
 		for _, pod := range podList.Items {
-			if strings.HasPrefix(pod.Name, "ovnkube-node") {
+			if strings.HasPrefix(pod.Name, "ovnkube-node") && pod.Spec.NodeName == testPod.Spec.NodeName {
```

Comment:
ideally this PR would be two separate commits, one fixing the existing test and one adding a new test